app/models/post.rb

```diff
@@ -0,0 +1,18 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comment
+  has_many :like
+
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :comments_counter, :likes_counter, comparison: { greater_than_or_equal_to: 0 }, numericality: true
+
```

Comment:
- [ ] OPTIONAL
It is a nice idea to use one of the Rails Active Callbacks `after_save` to call the `update_user_posts_counter` method every time a new post has been saved to database, then make the `update_user_posts_counter` method private, similar to:

```ruby
 # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

Reply:
Thanks